libndt.cpp

```diff
@@ -1123,13 +1134,89 @@ bool Client::resolve(const std::string &hostname,
       result = false;
       break;
     }
-    addrs->push_back(address); // we only care about address
+    addrs->push_back(address);  // we only care about address
     EMIT_DEBUG("- " << address);
   }
   this->freeaddrinfo(rp);
   return result;
 }
 
+// Utilities (SSL)
+
+int Client::maybessl_connect(const std::string &hostname, Socket fd,
+                             const sockaddr *sa, SockLen len) noexcept {
+  auto rv = this->connect(fd, sa, len);
+  if (rv == 0 && (impl->settings.proto & protocol::tls) != 0) {
+#ifdef HAVE_OPENSSL
+    SSL *ssl = nullptr;
+    {
+      SSL_CTX *ctx = ::SSL_CTX_new(SSLv23_client_method());
+      if (ctx == nullptr) {
+        EMIT_WARNING("SSL_CTX_new() failed");
+        this->closesocket(fd);
+        return -1;
+      }
+      if (impl->settings.ca_bundle_path.empty()) {
+        EMIT_WARNING("maybessl_connect: ca_bundle_path is empty");
+        ::SSL_CTX_free(ctx);
+        this->closesocket(fd);
+        return -1;
+      }
+      if (!SSL_CTX_load_verify_locations(  //
+              ctx, impl->settings.ca_bundle_path.c_str(), nullptr)) {
```

Comment:
We should provide, as an alternative mechanism the one where the CA bundle is a string. This is currently supported by OpenSSL, and it's unclear whether other SSL libraries support it.